crates/esplora/src/async_ext.rs

```diff
@@ -85,10 +85,11 @@ impl EsploraAsyncExt for esplora_client::AsyncClient {
         local_tip: CheckPoint,
         request_heights: impl IntoIterator<IntoIter = impl Iterator<Item = u32> + Send> + Send,
     ) -> Result<local_chain::Update, Error> {
-        let request_heights = request_heights.into_iter().collect::<BTreeSet<_>>();
         let new_tip_height = self.get_height().await?;
 
-        // atomically fetch blocks from esplora
+        // Atomically fetch latest blocks from Esplora. This way, we avoid creating an update with
+        // an inconsistent set of blocks (assuming that a reorg depth cannot be greater than the
+        // latest blocks fetched).
```

Comment:
Not touched as part of this PR but below:

```rust

        let mut fetched_blocks = {
            let heights = (0..=new_tip_height).rev();
            let hashes = self
                .get_blocks(Some(new_tip_height))
                .await?
                .into_iter()
                .map(|b| b.id);
            heights.zip(hashes).collect::<BTreeMap<u32, BlockHash>>()
        };
```
Why are we zipping to get the heights of the blocks. This looks really error prone. Surely esplora will tell us the heights of the blocks we fetched.

Reply:
But furthermore, why do we even do `get_height` at the start here. Just do `get_blocks(None)` which will get from the tip.

Reply:
I didn't realize there was a height available because it's nested in the `time` field.